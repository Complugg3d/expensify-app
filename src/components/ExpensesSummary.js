import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectTotalExpenses from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';



export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const expensesWord = expensesCount === 1 ? 'expense' : 'expenses';
  const formatedAmount = numeral(expensesTotal / 100).format('$0,0.00');

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">
          Viewing <span>{expensesCount}</span> {expensesWord} 
          totalling <span>{formatedAmount}</span>
        </h1>
        <div className="page-header__actions">
          <Link className="btn-layout" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectTotalExpenses(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
