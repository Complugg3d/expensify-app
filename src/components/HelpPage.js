import React from 'react';

const HelpPage = () => (
    <div>
        <p>4</p>
    </div>
);

export default HelpPage;