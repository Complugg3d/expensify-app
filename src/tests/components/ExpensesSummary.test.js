import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render expense Summary with expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={100}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should not render ExpensesSummary with empty message', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={123}/>);
    expect(wrapper).toMatchSnapshot();
});