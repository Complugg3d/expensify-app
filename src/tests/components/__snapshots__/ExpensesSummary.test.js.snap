// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render ExpensesSummary with empty message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expensesCount={1}
    expensesTotal={123}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Viewing ",
        1,
        " ",
        "expense",
        " totalling ",
        "$1.23",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Viewing ",
      1,
      " ",
      "expense",
      " totalling ",
      "$1.23",
    ],
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totalling ",
          "$1.23",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        1,
        " ",
        "expense",
        " totalling ",
        "$1.23",
      ],
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should render expense Summary with expenses 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expensesCount={3}
    expensesTotal={100}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Viewing ",
        3,
        " ",
        "expenses",
        " totalling ",
        "$1.00",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Viewing ",
      3,
      " ",
      "expenses",
      " totalling ",
      "$1.00",
    ],
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          3,
          " ",
          "expenses",
          " totalling ",
          "$1.00",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        3,
        " ",
        "expenses",
        " totalling ",
        "$1.00",
      ],
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
